tests/dummy/app/routes/guid.js

```diff
@@ -0,0 +1,26 @@
+import Ember from 'ember';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+
+export default Ember.Route.extend(AuthenticatedRouteMixin, {
+    model(params) {
+        return this.store.findRecord('guid', params.guid).then(g => g.get('referent'));
+    },
+    renderTemplate(controller, model) {
+        let kind = model.constructor.modelName;
+        if (kind === 'node') {
+            this.render('nodes.detail.index', {
```

Comment:
This is interesting- this function seems to have some pretty powerful options under the hood. Being able to specify view and controller to be used during rendering sounds like a plus.

I'll be interested to see how this affects things like action bubbling and `thingFor` (eg what the files detail page thinks its parent model/params are). Certainly worth a look.